### ISSUE TITLE
Vanity URL - Campaign Landing Page - Resources for Suicide Loss Survivors

### ISSUE THREAD

**Issue by TroyCMSSupport** [department-of-veterans-affairs/va.gov-team#90405]
**[2024-08-29] [Fran]**  Still awaiting word that the CLP is live, and then we'll be able to redirect.
**[2024-08-13]** CLP is not yet published: https://prod.cms.va.gov/initiatives/va-resources-for-survivors-of-suicide-loss

Related to https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18871

### Instructions
- Requests must be submitted **AT LEAST** 2 weeks in advance.  Some requests take a significant amount of time and coordination to implement, so start the process as soon as you know you will need one. 
- This issue will be used from initial request through implementation to ensure all individuals working on this are notified of status updates.  Please do not create multiple issues to track different steps.
- It is your responsibility to notify VA stakeholders as appropriate.

| POC | Github alias(es) |
--- | --- 
| Product team point(s) of contact: | @FranECross  |
| Content point of contact: | `CAIA will enter` |
| IA point of contact: | `IA will enter` |

### Type of request
- [ ] We are changing the URL of an existing page (complete redirect section)
- [x] We need a custom vanity URL (complete vanity URL section)

### Do you have a product technical team able to implement this redirect?
- [x] Yes - Please note: Redirect still needs approval from IA. -- Public Websites will implement
  - [x] Please verify your team has reviewed [Redirect governance](https://design.va.gov/components/url-standards/redirects) and [Redirect technical implementation information
](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md) 
- [ ] No - CAIA will need to make Public Websites aware of the ticket.

### Implementation date
When does this request need to be live? **NOTE:** Redirects in vsp-platform-revproxy auto-deploy M-Th. Friday launches should be avoided.
Choose one:

- [x] Any day prior to a deadline/latest possible date. ~~**8/26/2024**~~ as of 2024-08-29 still awaiting word that the CLP is live, and then we'll be able to redirect.

### Vanity URLs 
Vanity URL  |  Landing page
---  |  ---
https://va.gov/survivingsuicideloss | https://www.va.gov/initiatives/va-resources-for-survivors-of-suicide-loss

**Link to campaign landing page request issue, if relevant:**
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18871 

### Process, Roles and Responsibilities
- [x] Requesting team: Above information is provided
- [x] Requesting team: All appropriate VA stakeholders are notified as appropriate
- [x] IA: Request is vetted and documented and implementation plan is clear 
- [x] IA: Request is assigned to appropriate team for implementation 
- [ ] Implementation team: Work is complete
- [ ] Implementation team: Validated in production
- [ ] Requesting team: Validates everything is correct in production and closes ticket

**Comment by TroyCMSSupport** [comment]
@mnorthuis @ian-sears @jilladams - Please let me know if I have missed any pertinent information - Thank you!

**Comment by jilladams** [comment]
@kristinoletmuskat not sure if this ticket will notify you, since I edited assignees weird. FYI this exists, just in case, for IA review. 

(The CLP runbook template was linking to an older version of the redirect issue template with older assignees list. I've updated that template so it'll use latest version for any in future.)

**Comment by mnorthuis** [comment]
Hi @jilladams, vanity URL requests still come to me...I am working on this one and will respond this week.  Thanks!

**Comment by mnorthuis** [comment]
@TroyCMSSupport I see two different vanity URLs requested in the CLP request. Just want to make sure I have the right one. 

Vanity URL requested:
https://www.va.gov/resources-for-suicide-loss-survivors

Is there a preferred “Vanity URL” from the root of the [va.gov](http://va.gov/)website? [Va.gov/suicidelosssurvivors](http://va.gov/suicidelosssurvivors)

**Comment by TroyCMSSupport** [comment]
Thank you @mnorthuis - I will confirm and update

**Comment by TroyCMSSupport** [comment]
@mnorthuis  - the correct vanity URL should be va.gov/suicidelosssurvivors

**Comment by mnorthuis** [comment]
Vanity URL request is approved, you can move forward with setting it up/redirecting.  Thank you!

**Comment by FranECross** [comment]
@mnorthuis and @TroyCMSSupport Shouldn't the landing page be a VA.gov page, and not prod.cms.va.gov? Thanks! cc @jilladams

**Comment by mnorthuis** [comment]
@FranECross Yes, sorry! Updated.  Since the CLP is not live yet, this should be double checked to make sure the URL didn't change before implementing.  I can assume this is the production URL, but they could change things before launch.

**Comment by FranECross** [comment]
@mnorthuis Thanks! Will we be pinged when the CLP is live to ensure we coordinate correctly? (I'm assuming so, but crossing my Ts here. 😉)

**Comment by mnorthuis** [comment]
@TroyCMSSupport I'm not sure who is the POC with Justin on the CLP request, but if someone can let him know that the URLs have been finalized, then Fran and team can set up the vanity URL/redirect whenever they are ready - it will of course 404 until the page is published, but it will be there once they do.

**Comment by ian-sears** [comment]
@jilladams 
Justin Warren requesting to change vanity url to https://www.va.gov/survivingsuicideloss
can you let the @ian-sears and @TroyCMSSupport know when this is turned around or complete.
@mnorthuis @RLHecht @strelichl for awareness

**Comment by jilladams** [comment]
@mnorthuis any concerns about changing the vanity URL for this from `/suicidelosssurvivors` to `/survivingsuicideloss`?

**Comment by mnorthuis** [comment]
That changes the context of the content to being about surviving the event itself vs providing resources and benefits for those that have lost someone to suicide. I reached out to Justin to let him know the potential impact of that change in context. Waiting for a response.

**Comment by mnorthuis** [comment]
Justin indicated that the SMEs are making some content changes that will likely bring the CLP in line with the context of the newly requested vanity URL, so I am fine with the change.

**Comment by FranECross** [comment]
@TroyCMSSupport Will you please update to the correct URL in the Vanity URLs section of the ticket? I believe it's supposed to be: https://www.va.gov/survivingsuicideloss

**Comment by jilladams** [comment]
I went ahead and updated that.

**Comment by ian-sears** [comment]
@jilladams @mnorthuis @FranECross 
Verifying that the Vanity URL will work when this is Published. Justin Warren is asking just now at the CMS Help Desk.
THe "Generate auto URL" node checkbox in the node's URL alias field is "checked".
The node's URL alias is reflecting the title, which may have changed as they were editing this over the last couple of weeks.
Currently reads (in the node): /initiatives/va-resources-for-survivors-of-suicide-loss
(this is inconsistent with this GitHub ticket's original change request, top of this ticket)

Prior to me changing anything, can I ask for your eyes on this (high profile CLP) which may be about to launch?

**Comment by jilladams** [comment]
Original CLP node URL / landing URL was: https://www.va.gov/initiatives/resources-for-suicide-loss-survivors
And is now: https://www.va.gov/initiatives/va-resources-for-survivors-of-suicide-loss

I updated the link in the body of the ticket. 

We can go ahead and create this redirect. It will 404 until the node is published, but that's not a problem since no PR will be in the wild advertising the vanity URL til the CLP publishes anyway. (Basically: there's no traffic going to anything, so a 404 doesn't matter and won't be an issue once they publish.) Since we expect this to publish soon, I think we should go ahead and have someone handle it @FranECross -- I pulled it into sprint.

**Comment by jilladams** [comment]
@ian-sears noted [in Slack](https://dsva.slack.com/archives/CT4GZBM8F/p1725557290345719?thread_ts=1725556793.215079&cid=CT4GZBM8F) that this CLP is planning to publish on Monday 9/9 and we'll want the redirect live by then. Meaning: it needs to merge by tomorrow EOD to make it.

**Comment by jilladams** [comment]
Notified Justin at VHA of deploy timing on the helpdesk ticket: https://va-gov.atlassian.net/browse/VAHELP-7876?focusedCommentId=50381

@jv-agile6 please notify me and Fran once the PR is merged. Thanks!

**Comment by jv-agile6** [comment]
Moved ticket to in-progress.

**Comment by jilladams** [comment]
@laflannery FYI re: Monday verification

**Comment by jv-agile6** [comment]
```# September 6, 2024
# https://github.com/department-of-veterans-affairs/va.gov-team/issues/90405
  - src: ^/survivingsuicideloss/?*
    dest: /initiatives/va-resources-for-survivors-of-suicide-loss/
    matcher: "~*"```

This YAML configuration defines a redirect rule for web server configurations. Let’s break down each component:

src: This is the source URL pattern that the redirect rule will match against.

^/survivingsuicideloss/?*:
^ asserts the position at the start of the URL path.
/survivingsuicideloss/ matches the specific path segment.
/? makes the trailing slash optional (i.e., it matches both /survivingsuicideloss and /survivingsuicideloss/).
* means that it can match any characters that follow, allowing for more flexible matching.
dest: This is the destination URL where the source URL should be redirected to.

/initiatives/va-resources-for-survivors-of-suicide-loss/ is the new URL where users will be redirected.
matcher: This specifies how the source URL pattern should be matched.

~* is often used to indicate a case-insensitive match in some regex-based systems. However, its exact behavior can vary depending on the specific system you're using.
In summary, this rule redirects any URL that starts with /survivingsuicideloss (and optionally followed by more characters) to /initiatives/va-resources-for-survivors-of-suicide-loss/. The matcher: "~*" part suggests a case-insensitive match for the source URL pattern.

**Comment by jv-agile6** [comment]
Merged https://github.com/department-of-veterans-affairs/vsp-platform-revproxy/pull/613 moving ticket to QA

**Comment by jilladams** [comment]
Bumping over to Prod verification - the only test we can do is in prod once it deploys, to verify the redirect is working. LauraF is enlisted to help with that on Monday after the revproxy deploys. 

Revproxy daily deploy is here: https://github.com/department-of-veterans-affairs/vsp-platform-revproxy/actions/workflows/deploy-daily.yaml, and looks like the last few jobs have kicked off at ~10:03ET and completed in 28-35 mins, so we should be able to verify by ~10:40a ET / ~9:40a CT.

@laflannery @jv-agile6 The test will be: 
* click https://va.gov/survivingsuicideloss
* browser should redirect to https://www.va.gov/initiatives/va-resources-for-survivors-of-suicide-loss
   * which will 404 until JustinW publishes the CLP

**Comment by laflannery** [comment]
@jilladams and @jv-agile6 I just tested this and the redirect isn't working for me, the vanity URL is remaining in my address bar:
![image](https://github.com/user-attachments/assets/d37ad7a5-d9bf-491c-8922-dd75059317f4)

**Comment by laflannery** [comment]
I just tried again and this is working now:
![image](https://github.com/user-attachments/assets/671291cb-4661-470e-9111-cae2550f7fdc)

**Comment by jilladams** [comment]
Since this came in from our own house via CLP request, I'm going to go ahead and close. We can reopen if we find anything amiss once the CLP publishes.

**Comment by ian-sears** [comment]
FYSA all seems ready and working 
@department-of-veterans-affairs/platform-cms-helpdesk @jilladams @FranECross @mnorthuis 
The Vanity URL now resolves to the correct URL (tested and verified this morning).
http://www.va.gov/survivingsuicideloss
https://www.va.gov/initiatives/va-resources-for-survivors-of-suicide-loss
https://prod.cms.va.gov/initiatives/va-resources-for-survivors-of-suicide-loss
